modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -83,6 +93,19 @@ public SourceHandler(SourceConfiguration sourceConfiguration) {
             this.threadingView = new ThreadingView(PassThroughConstants.PASSTHOUGH_HTTP_SERVER_WORKER, true, 50);
         }
 
+        Properties props = MiscellaneousUtil.loadProperties(PROPERTY_FILE);
+        String validationProperty = MiscellaneousUtil.getProperty(props, MESSAGE_SIZE_VALIDATION, "false");
+        String validMaxMessageSizeStr = MiscellaneousUtil
+                .getProperty(props, VALID_MAX_MESSAGE_SIZE, String.valueOf(Integer.MAX_VALUE));
+        isMessageSizeValidationEnabled = Boolean.valueOf(validationProperty);
+        try {
+            validMaxMessageSize = Integer.valueOf(validMaxMessageSizeStr);
+        } catch (NumberFormatException e) {
+            log.warn("Invalid max message size configured for property \"valid.max.message.size.in.bytes\", "
```

Comment:
is there any reason for not throwing the exception here ? also need to include the exception stacktrace ?